lib/editor/components/pattern/NormalizeStopTimesModal.js

```diff
@@ -56,14 +60,28 @@ export default class NormalizeStopTimesModal extends Component<Props, State> {
             value={this.state.patternStopIndex}
             componentClass='select'
             onChange={this._onChangeStop}>
-            {activePattern.patternStops.map((patternStop, index) => {
-              const stop = stops.find(s => s.stop_id === patternStop.stopId)
-              if (!stop) return null
+            {patternHalts.map((patternHalt, index) => {
+              let displayName
+              if (patternHaltIsStop(patternHalt)) {
+                // $FlowFixMe: Flow doesn't recognize that patternHaltIsStop is a type check
```

Comment:
For another PR: one way to remove these validation messages in this and other files is to create a function that extracts a `PatternStop`
```js
function getPatternStop(patternHalt: PatternHalt): PatternStop {
  // return patternHalt if it is a pattern stop, null otherwise
}
...
const patternStop = getPatternStop(patternHalt)
if (patternStop) {
  ...
}
```
and same for the other stop types.

Reply:
That's a good idea. I'll add a TODO for this.